internal/serverinstall/nomad.go

```diff
@@ -813,29 +839,48 @@ func waypointNomadJob(c nomadConfig, rawRunFlags []string) *api.Job {
 	grpcPort, _ := strconv.Atoi(defaultGrpcPort)
 	httpPort, _ := strconv.Atoi(defaultHttpPort)
 
-	// Include services to be registered in Consul. Currently configured to happen by default
+	// Include services to be registered. Currently configured to happen with Consul by default
 	// One service added for Waypoint UI, and one for Waypoint backend port
-	if c.consulService {
+	var services []*api.Service
+	if (c.serviceProvider == "consul" && (c.consulService || upgrade)) || (c.consulService && (c.serviceProvider == "consul" || upgrade)) {
```

Comment:
Would it be worth while to emit a warning if we find that a user is using the `-nomad-consul-service` flag instead of the `-nomad-service-provider` flag?

Reply:
Oh, i see, its above :D ignore me!